src/webui/webapplication.h

```diff
@@ -60,13 +63,19 @@ class WebSession final : public ISession
     bool hasExpired(qint64 seconds) const;
     void updateTimestamp();
 
-    QVariant getData(const QString &id) const override;
-    void setData(const QString &id, const QVariant &data) override;
+    template <typename T>
+    void registerAPIController(const QString &scope)
+    {
+        static_assert(std::is_base_of_v<APIController, T>, "Type should be derived from APIController.");
+        m_apiControllers[scope] = new T(this);
+    }
+
+    APIController *getAPIController(const QString &scope) const;
 
 private:
     const QString m_sid;
     QElapsedTimer m_timer;  // timestamp
-    QVariantHash m_data;
+    QHash<QString, APIController *> m_apiControllers;
```

Comment:
I'm thinking of if it is possible to provide faster lookup... since the strings are fixed and have different initials, I'm thinking maybe `QMap` would be faster/appropriate here.